app/controllers/api/storage.php

```diff
@@ -21,8 +22,106 @@
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'createBucket')
+    ->label('sdk.description', '/docs/references/storage/create-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET)
+    ->param('name', '', new Text(128), 'Bucket name', false)
+    ->param('maximumFileSize', 0, new Integer(), 'Maximum file size supported', true)
+    ->param('allowedFileExtensions', ['*'], new ArrayList(new Text(64)), 'Allowed file extensions', true)
+    ->param('enabled', true, new Boolean(), 'Bucket enabled', true)
+    ->param('adapter', 'local', new WhiteList(['local']), 'Storage adapter', true)
+    ->param('encryption', true, new Boolean(), 'encryption is enabled', true)
+    ->param('antiVirus', true, new Boolean(), 'Virus scanning is enabled', true)
+    ->param('read', null, new ArrayList(new Text(64)), 'An array of strings with read permissions. By default only the current user is granted with read permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->param('write', null, new ArrayList(new Text(64)), 'An array of strings with write permissions. By default only the current user is granted with write permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('user')
+    ->inject('audits')
+    ->action(function ($name, $maximumFileSize, $allowedFileExtensions, $enabled, $adapter, $encryption, $antiVirus, $read, $write, $response, $dbForInternal, $user, $audits) {
+        /** @var Utopia\Swoole\Request $request */
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Database\Document $user */
+        /** @var Appwrite\Event\Event $audits */
+
+        try {
+            $data = [
+                '$collection' => 'buckets',
+                'dateCreated' => \time(),
+                'dateUpdated' => \time(),
+                'name' => $name,
+                'maximumFileSize' => $maximumFileSize,
+                'allowedFileExtensions' => $allowedFileExtensions,
+                'enabled' => $enabled,
+                'adapter' => $adapter,
+                'encryption' => $encryption,
+                'antiVirus' => $antiVirus,
+            ];
+
+            $data['$read'] = (is_null($read) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $read ?? []; //  By default set read permissions for user
+            $data['$write'] = (is_null($write) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $write ?? []; //  By default set write permissions for user
+            $data = $dbForInternal->createDocument('buckets', new Document($data));
+        } catch (AuthorizationException $exception) {
```

Comment:
We don't need all this try-catch statements anymore with the new DB library. Follow other `createDocument` we have in the code. The DB controller is not yet implemented, so it's better to use another example.

We usually use all this check for better error handling when we create custom documents, on an internal object where we control the structure it's not needed.

Reply:
removed